lib/workspace.js

```diff
@@ -396,7 +447,20 @@ module.exports = {
       if (localPathExists) {
         debug(`Syncing ${localPath} to ${remotePath}`);
         await fse.copy(localPath, sourcePath);
+      }
+
+      if (dynamicFiles) {
+        await async.eachSeries(dynamicFiles, async ({ name, localPath, buffer }) => {
+          await fse.ensureDir(path.join(sourcePath, path.dirname(name)));
+          if (localPath) {
+            await fse.copy(localPath, path.join(sourcePath, name));
+          } else {
+            await fse.writeFile(path.join(sourcePath, name), buffer);
+          }
```

Comment:
As above, we should ensure that users can't create or write to arbitrary files, which in this case means validating that the results of `path.join(sourcePath, ...)` don't resolve to somewhere outside of `sourcePath`.

Reply:
I believe this is resolved in 4f38c71.